src/Orleans/Streams/Predicates/FilterPredicateWrapperData.cs

```diff
@@ -50,6 +50,9 @@ internal class FilterPredicateWrapperData : IStreamFilterPredicateWrapper
         [NonSerialized]
         private StreamFilterPredicate predicateFunc;
 
+        // constructor used by serializator
+        private FilterPredicateWrapperData() { }
```

Comment:
As for references to 2 JSON serializers: that is exactly the problem (in my view) with json serializers. It has to put all kind of restrictions and rules on the serialized classes. Newtonsoft specifically is quite flexible, it allows like 5 different ways to integrate with serializer (public properties, special Json attributes, ISerializable, json converters, ...) but still - **it does not JUST WORK**.
It would be nice to main tain the same "it JUST WORKs" for Orleans serializer, unless it is not possible otherwise.

Also, I did not quite understand: would this be a restriction only for Orleans on CooreClr or also for Orleans on full .NET?